src/main/java/org/tarantool/TarantoolRequest.java

```diff
@@ -0,0 +1,163 @@
+package org.tarantool;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class TarantoolRequest implements Comparable<TarantoolRequest> {
```

Comment:
okay, now we have separated requests and operations. The requests are used as a reusable container to be executed. The operations are an internal state of the current request executions.
Aldo, I removed `TarantoolOp` class because of its unnecessary and returned JDK `CompletableFuture` instead.